File: PersistenceAndServices/src/main/java/repository/UserDatabaseRepository.java
package repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import domain.User;
import repository.utils.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;
import java.util.Properties;

public class UserDatabaseRepository implements UserRepository{
    private JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger();

    public UserDatabaseRepository (Properties props) {
        logger.info("Initialized UserDatabaseRepository with properties: {}",props);
        dbUtils = new JdbcUtils(props);
    }

    @Override
    public String getPasswordOfUser (String username) throws Exception {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        try {
            PreparedStatement statement = con.prepareStatement
                    ("SELECT password FROM Users WHERE username = ?");
            statement.setString(1,username);
            ResultSet result = statement.executeQuery();
            result.next();

            String hashedPassword = result.getString("password");
            result.close();
            logger.traceExit(hashedPassword);
            return hashedPassword;
        }
        catch (SQLException ex) {
            logger.error(ex);
            throw new Exception("User not found!\n");
        }
    }

    @Override
    public User find(Integer integer) throws Exception {
        return null;
    }

    @Override
    public List<User> getAll() throws Exception {
        return null;
    }

    @Override
    public void add(User entity) throws Exception {

    }

    @Override
    public User delete(Integer integer) throws Exception {
        return null;
    }

    @Override
    public User update(User entity) throws Exception {
        return null;
    }

    @Override
    public Integer size() throws Exception {
        return null;
    }

}

File: MPPCS/Repository/UserDatabaseRepository.cs
using Persistence.Repository.Utils;
using CommonDomain;
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Persistence.Repository
{
    public class UserDatabaseRepository : UserRepository
    {
        private DbUtils dbUtils;
        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);

        public UserDatabaseRepository( string url )
        {
            dbUtils = new DbUtils(url);
        }
        public string GetPasswordOfUser(string username)
        {
            log.Debug("acquiring password of " + username);
            SQLiteConnection con = dbUtils.GetConnection();
            try
            {
                SQLiteCommand command = con.CreateCommand();
                command.CommandText = "SELECT password FROM Users WHERE username=@1";
                command.Parameters.Add(new SQLiteParameter("@1", username));
                SQLiteDataReader dataReader = command.ExecuteReader();
                dataReader.Read();
                string password = dataReader.GetString(0);
                dataReader.Close();
                log.Debug("FOUND " + password);
                return password;
            }
            catch (Exception ex)
            {
                log.Error(ex);
                throw new Exception("Invalid login!");
            }
        }
        public void Add(User entity)
        {
            throw new NotImplementedException();
        }

        public User Delete(int id)
        {
            throw new NotImplementedException();
        }

        public User Find(int id)
        {
            throw new NotImplementedException();
        }

        public List<User> GetAll()
        {
            throw new NotImplementedException();
        }

        public int Size()
        {
            throw new NotImplementedException();
        }

        public User Update(User entity)
        {
            throw new NotImplementedException();
        }
    }
}
